### DISCUSSION TITLE
how does Docker use count towards the device limit

### DISCUSSION THREAD

**Discussion by MichalDN** [dashingsoft/pyarmor#1744] [Q&A] [answered]
We just purchased a pro license for obfuscating packages of our product (one product made of 10 packages).
Using two build pcs, (linux/windows), we build these packages.
on windows, it is without docker.
on linux we use a cibuildwheel, which uses a docker image to build it. (runs, finishes shuts down) and then identical is started again.
I just noticed that, if I run the docker image interactively, the machine id changes on every execution of pyarmor reg -g 1

(venv) [root@e9be3027b9af tmp]# pyarmor reg -g 1
INFO     current machine id is "m4bc5ef60da3d4f2f51cd3eafcbaf75f0"
INFO     current machine id is "m9e6e47b14ac82acf9cceaae30f214d07"
INFO     current machine id is "mf91791c654ff50bb700e4f79930a612c"
INFO     current machine id is "mad97dc11cc9398350178025964330ab7"
...

Does this count towards the "100 different devices are allowed to obfuscate the scripts for each license."?

We didn't need the offline mode, and at most 1 container runs at the same time, so this did not seem like a Group use case.

Any information will be greatly appreciated.

**Comment by jondy** [comment]
Each docker run will be counted if docker container machine id is changed.
Only if it's less than 100 times in one day (24-hours), pro license is OK.

**Reply by MichalDN** [reply]
I see, thank you for confirmation. One thing that confuses me however. 

If I start the docker in interactive mode (-it), is it expected behavior that each call of `pyarmor reg -g 1` will return a different machine id? While in the same console? That feels unexpected.

**Comment by jondy** [comment, marked as answer]
For this case the solution at this time is to use group license and run `pyarmor-auth` in docker host.
I don't know how to check docker host information in docker container for interactive mode, if it's possible, in later version, it may be taken host machine id as docker container id.

**Reply by MichalDN** [reply]
Not the answers I hoped for, but it is the answer need. Thank you for your time :)
